### PR TITLE
[feature](merge-cloud) Fix cloud cluster checker cant get backends

### PR DESCRIPTION
**Pull request by deardeng**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by deardeng** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java [hunk 1]
```diff
@@ -301,9 +302,12 @@ protected void runAfterCatalogReady() {
...
                     + "cloud_unique_id={}, response={}", Config.cloud_unique_id, response);
         } else {
-            // clusterId -> clusterPB
-            Map<String, ClusterPB> remoteClusterIdToPB = new HashMap<>();
             Set<String> localClusterIds = clusterIdToBackend.keySet();
+            // clusterId -> clusterPB
+            Map<String, ClusterPB> remoteClusterIdToPB = response.getClusterList().stream()
+                    .filter(c -> c.getType() != Type.SQL)
+                    .collect(Collectors.toMap(ClusterPB::getClusterId, clusterPB -> clusterPB));
+            LOG.info("get cluster info  clusterIds: {}", remoteClusterIdToPB);
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java:310]: th word "clusterIds" is redundant

Comment (deardeng) [reply fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java]: fixed

**Comment by deardeng** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java [hunk 2]
```diff
@@ -291,78 +292,11 @@ private void checkDiffNode(Map<String, ClusterPB> remoteClusterIdToPB,
...
-                backendAliveTotal.addLabel(new MetricLabel("cluster_id", entry.getValue()));
-                backendAliveTotal.addLabel(new MetricLabel("cluster_name", key));
-                MetricRepo.DORIS_METRIC_REGISTER.addMetrics(backendAliveTotal);
-                return backendAliveTotal;
-            }).setValue(aliveNum);
-        }
-
+        getCloudBackends();
+        updateCloudMetrics();
         LOG.info("daemon cluster get cluster info succ, current cloudClusterIdToBackendMap: {}",
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java:297]: this log should be in getcloudbackends()

File: fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java [hunk 3]
```diff
@@ -470,5 +404,81 @@ private void getObserverFes() {
...
+    private void getCloudBackends() {
+        Map<String, List<Backend>> clusterIdToBackend = Env.getCurrentSystemInfo().getCloudClusterIdToBackend();
+        //rpc to ms, to get mysql user can use cluster_id
+        // NOTE: rpc args all empty, use cluster_unique_id to get a instance's all cluster info.
+        Cloud.GetClusterResponse response = CloudSystemInfoService.getCloudCluster("", "", "");
+        if (!response.hasStatus() || !response.getStatus().hasCode()
+                || (response.getStatus().getCode() != Cloud.MetaServiceCode.OK
+                && response.getStatus().getCode() != MetaServiceCode.CLUSTER_NOT_FOUND)) {
+            LOG.warn("failed to get cloud cluster due to incomplete response, "
+                    + "cloud_unique_id={}, response={}", Config.cloud_unique_id, response);
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java:417]: this function should return here.

File: fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java [hunk 4]
```diff
@@ -470,5 +404,81 @@ private void getObserverFes() {
...
+        } else {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java:418]: eliminate `else` block to reduce indention

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by deardeng** [comment]
run buildadd

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Pull request merged**